### PR TITLE
[Enhancement] fix aws sdk use

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

We have to put environment variable initilization before `export_env_from_conf`, otherwise we can not override some environment variables.
```
# https://github.com/aws/aws-cli/issues/5623
# https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html
export AWS_EC2_METADATA_DISABLED=true

export_env_from_conf $STARROCKS_HOME/conf/be.conf
export_mem_limit_from_conf $STARROCKS_HOME/conf/be.conf
```

Plus, I add a config value `aws_sdk_logging_trace_enabled`.  By enabling this, we can see trace log by aws-sdk-cpp and diagnose problems.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/common/config.h [hunk 1]
```diff
@@ -721,6 +721,8 @@ CONF_Int32(io_coalesce_read_max_distance_size, "1048576");
 
 CONF_Int32(connector_io_tasks_per_scan_operator, "16");
 
+CONF_mBool(aws_sdk_logging_trace_enabled, "false");
```
Comment (DorianZheng) [inline be/src/common/config.h:724]: add comment

Comment (DorianZheng) [reply be/src/common/config.h:724]: https://docs.aws.amazon.com/zh_cn/sdk-for-cpp/v1/developer-guide/logging.html

**Review by DorianZheng — DISMISSED** [review]
left a comment, rest LGTM

File: be/src/common/config.h (hunk 1 above)
Comment (dirtysalt) [reply be/src/common/config.h:724]: @DorianZheng you mean add the link as comment into the code?

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Review by DorianZheng — APPROVED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by satanson — APPROVED** [review]

**Comment by dirtysalt** [comment]
@MergifyIO backport branch-2.3

**Comment by dirtysalt** [comment]
@MergifyIO backport branch-2.4

**Pull request merged**